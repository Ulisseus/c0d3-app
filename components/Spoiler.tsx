import * as React from 'react'
type Props = {
  name?: string
  offset?: string
}
const Spoiler: React.FC<Props> = ({ children, name, offset }) => {
  const [status, setStatus] = React.useState({
    css: { display: 'none' },
    name: 'spoiler_arrow__down'
  })
  const onClick = () => {
    status.css.display === 'none'
      ? setStatus({ css: { display: 'inherit' }, name: 'spoiler_arrow__up' })
      : setStatus({ css: { display: 'none' }, name: 'spoiler_arrow__down' })
  }
  return (
    <>
      <div
        className="mb-1"
        style={{ marginLeft: offset ? offset : '1em' }}
        onClick={onClick}
      >
        {name ? name : 'Answer'}
        <div className={status.name} />
      </div>
      <div style={status.css}>{children}</div>
    </>
  )
}

export default Spoiler
