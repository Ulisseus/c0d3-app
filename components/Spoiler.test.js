import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Spoiler from './Spoiler'

describe('Spoiler component test', () => {
  test('Should render spoiler', async () => {
    const { container } = render(
      <Spoiler name="TestSpoiler">
        <h1>Hidden text</h1>
      </Spoiler>
    )
    expect(container).toMatchSnapshot()
    userEvent.click(screen.getByText('TestSpoiler'))
    expect(screen.getByText('Hidden text')).toBeTruthy
    userEvent.click(screen.getByText('TestSpoiler'))
  })
  test('Should render spoiler without name', () => {
    render(
      <Spoiler>
        <h1>Hidden text</h1>
      </Spoiler>
    )
    expect(screen.getByText('Answer')).toBeTruthy()
  })
  test('Should render spoiler with offset', () => {
    const { container } = render(
      <Spoiler offset="1em">
        <h1>Hidden text</h1>
      </Spoiler>
    )
    expect(container).toMatchSnapshot()
  })
})
