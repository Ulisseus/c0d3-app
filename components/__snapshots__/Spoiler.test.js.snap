// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spoiler component test Should render spoiler 1`] = `
<div>
  <div
    class="mb-1"
    style="margin-left: 1em;"
  >
    TestSpoiler
    <div
      class="spoiler_arrow__down"
    />
  </div>
  <div
    style="display: none;"
  >
    <h1>
      Hidden text
    </h1>
  </div>
</div>
`;

exports[`Spoiler component test Should render spoiler with offset 1`] = `
<div>
  <div
    class="mb-1"
    style="margin-left: 1em;"
  >
    Answer
    <div
      class="spoiler_arrow__down"
    />
  </div>
  <div
    style="display: none;"
  >
    <h1>
      Hidden text
    </h1>
  </div>
</div>
`;
